### PR TITLE
Configure Frontend CI

### PR DESCRIPTION
**Pull request by r-aya**
## Notion ticket link

Configure Frontend CI (https://www.notion.so/uwblueprintexecs/Configure-CI-frontend-7a9ddfb6bb1e4732990c393ebc06bed0)

## Implementation description
* Configured frontend CI
* Linted frontend directory

## Steps to test
1. Verify that workflows correctly

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: .github/workflows/ci-frontend.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  # Triggers the workflow on push or pull request events but only for the development branch
+  push:
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "frontend_ci"
+  frontend_ci:
```
Comment (Tahmeed27) [inline .github/workflows/ci-frontend.yml:14]: could you rename this frontend_lint just to be uniform with the other ticket

File: frontend/src/APIClients/AuthAPIClient.ts [hunk 2]
```diff
@@ -1,8 +1,3 @@
-import {
-  FetchResult,
-  MutationFunctionOptions,
-  OperationVariables,
-} from "@apollo/client";
```
Comment (Tahmeed27) [inline frontend/src/APIClients/AuthAPIClient.ts:5]: Why was this removed? Was it an unused import?

File: .github/workflows/ci-frontend.yml (hunk 1 above)
Comment (TheRealJessicaLi) [reply .github/workflows/ci-frontend.yml:14]: Done, thanks!

File: frontend/src/APIClients/EntityAPIClient.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ const getFile = async (uuid: string): Promise<string> => {
       headers: { Authorization: bearerToken },
     });
 
-    return data.fileURL
+    return data.fileURL;
```
[the remark this answers is not in the document]
Comment (TheRealJessicaLi) [reply frontend/src/APIClients/EntityAPIClient.ts:74]: I'm not even sure LOL, good thing the linter caught it

File: .github/workflows/ci-frontend.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/checkout@v2
+      
+      # Install dependencies
+      - name: Install dependencies
+        run:  yarn --cwd ./frontend
+        
+      #Lint Check
+      - name: Run lint for frontend
+        working-directory: ./frontend
+        run: yarn lint
```
[the remark this answers is not in the document]
Comment (TheRealJessicaLi) [reply .github/workflows/ci-frontend.yml]: Oh jk that doesn't work, we need to install yarn lint first

**Review by Tahmeed27 — APPROVED** [review]
LGTM :) Remember to squash your commits when merging into dev

File: frontend/src/APIClients/AuthAPIClient.ts (hunk 2 above)
Comment (Tayef-Shah) [reply frontend/src/APIClients/AuthAPIClient.ts:5]: hmm apollo client? Is that for GraphQL?

Comment (Tahmeed27) [reply frontend/src/APIClients/AuthAPIClient.ts:5]: Yep it is lol
